stm/stm.ml

```diff
@@ -1320,12 +1286,18 @@ end = struct (* {{{ *)
 
   let get_prev_proof ~doc id =
     try
-      let np = get_proof ~doc id in
-      match np with
+      let np = ref None in
+      let vid = ref id in
+      (* handle "Show" command, which passes (last valid id)+1 *)
+      while !np = None && Stateid.to_int !vid > 0 do
+        np := get_proof ~doc !vid;
+        vid := Stateid.of_int (Stateid.to_int !vid - 1)
+      done;
+      match !np with
```

Comment:
I think this code is incorrect; this is not the way to traverse the document, as ids may not be linear.

Reply:
Fixed